ipset_test.go

```diff
@@ -244,25 +244,45 @@ func TestIPSetIntersection(t *testing.T) {
 		intersection []string
 	}{
 		{
-			[]string{"10.0.0.0/20", "10.5.8.0/24", "10.23.324.100/23"},
-			[]string{"10.0.0.2/30", "10.5.8.200/29", "10.23.324.104/27"},
-			[]string{"10.0.0.2/30", "10.5.8.200/29", "10.23.324.104/27"},
+			[]string{"10.0.20.0/20", "10.5.8.0/24", "10.23.224.100/23"},
+			[]string{"10.0.20.2/30", "10.5.8.200/29", "10.23.224.104/27"},
+			[]string{"10.23.224.104/27", "10.0.20.2/30", "10.5.8.200/29"},
 		},
 		{
-			[]string{"10.0.0.2/30", "10.5.8.200/29", "10.23.324.104/27"},
-			[]string{"10.0.0.0/20", "10.5.8.0/24", "10.23.324.100/23"},
-			[]string{"10.0.0.2/30", "10.5.8.200/29", "10.23.324.104/27"},
+			[]string{"10.10.0.2/30", "10.5.8.200/29", "10.23.224.104/27"},
+			[]string{"10.10.0.0/20", "10.5.8.0/24", "10.23.224.100/23"},
+			[]string{"10.10.0.2/30", "10.5.8.200/29", "10.23.224.104/27"},
 		},
 		{
-			[]string{"10.0.5.0/24", "10.5.8.200/29", "10.23.324.104/27"},
-			[]string{"10.6.0.0/24", "10.9.9.0/29", "10.23.6.100/23"},
+			[]string{"10.0.5.0/24", "10.5.8.200/29", "10.23.224.104/27"},
+			[]string{"10.6.0.0/24", "10.9.9.1/29", "10.23.6.100/23"},
```

Comment:
My opinion is that ipset shouldn't accept any CIDR where the host part is non-zero. IOW, adding 10.9.9.1/29 should be an error and this should fail. Take a look at the IPAM code in RIAS for an example of where I check that the host part is zero.

The more I think about it, the less it makes sense. These CIDRs represent sets of IPs in CIDR form. All of the IPs contained in the CIDR should be added to the set. For example, if the CIDR is 10.9.9.0/29, then 8 IP addresses are added to the set: 10.9.9.0 - 10.9.9.7 inclusive. With this in mind, it doesn't really make sense for the CIDR to be 10.9.9.1 because that is just one of the 8 IPs and isn't really special. Making sense?

I discussed this with Henry the other day and he and I seemed to be in agreement so maybe he could back me up here. :)

Reply:
Ok, I get it now, So the test cases should be CIDRs with host part all zeroes.